flamingo/scripts/hot_gas_plots.py

```diff
@@ -0,0 +1,460 @@
+
+    # Every other element
+    for i in range(len(all_elements) - 1):
+        ne_nH += elements_array[:, i + 1] * (element_masses[0] / element_masses[i + 1]) * (agtomic_num[i + 1] / agtomic_num[0])
+        ni_nH += elements_array[:, i + 1] * (element_masses[0] / element_masses[i + 1]) 
+        mu += elements_array[:, i + 1] / (agtomic_num[0] + agtomic_num[i + 1])
+
+    return ne_nH, ni_nH, mu
+
+def read_hot_gas_props(run_directory, snapshot_name, cat_name, obs_data_dir, chunk):
+    
+    elements = ['hydrogen', 'helium', 'carbon', 'nitrogen', 'oxygen', 'neon', 'magnesium', 'silicon', 'iron']
+    all_elements = ['hydrogen', 'helium', 'carbon', 'nitrogen', 'oxygen', 'neon', 'magnesium', 'silicon', 'sulphur', 'calcium', 'iron']
+    SulphurOverSilicon = 0.6054160
+    CalciumOverSilicon = 0.0941736
+
+    catalogue = load_catalogue(run_directory + '/' + cat_name)
+    groups = load_groups(run_directory + '/' + cat_name.replace('properties', 'catalog_groups'), catalogue=catalogue)
+
+    df_chunk = pd.DataFrame()
+    halo_n = 1
+    tot_h = len(chunk.index)
+    for index, row in chunk.iterrows():
+        print('TaskID: ', multiprocessing.current_process()._identity[0], 'Processing halo = ', int(row.GroupNumber), '(', halo_n, '/', tot_h, ')')
+        particles, unbound_particles = groups.extract_halo(halo_id=int(row.GroupNumber))
+        data, bmask = to_swiftsimio_dataset(particles, run_directory + '/' + snapshot_name, generate_extra_mask=True)
+        udata, umask = to_swiftsimio_dataset(unbound_particles, run_directory + '/' + snapshot_name, generate_extra_mask=True)
+        mask = np.logical_or(bmask.gas, umask.gas)
+
+        meta = data.metadata        
+        BoxSize = meta.boxsize[0].value
+        cosmo =  meta.cosmology
+        aexp = meta.a
+        z = meta.redshift
+        ez = cosmo.efunc(z)
+
+        df_elements = pd.DataFrame()
+        for element in elements:
+            df_elements[element] = getattr(data.gas.smoothed_element_mass_fractions, element)
+
+        df_gas = pd.DataFrame(data.gas.coordinates.value, columns=['x', 'y', 'z'])
+        df_gas['temp'] = data.gas.temperatures.value
+        df_gas['mass'] = data.gas.masses.to('Msun').value.astype(np.float64) * u.Msun.to(u.gram)
+        # df_gas['metallicity'] = data.gas.metal_mass_fractions.value / 0.0134
+        df_gas['sfr'] = data.gas.star_formation_rates.to('Msun/Gyr').value
+        df_gas['rho'] = data.gas.densities.to('g * cm ** -3').value
+        df_gas['nH'] = df_elements.hydrogen.values * df_gas.rho.values / mp 
+        # df_gas['nH'] = np.full_like(df_gas.temp.values, 1e6)
+        df_gas['lambda_c'] = np.power(10, xray.interpolate_X_Ray_pandas(np.log10(df_gas.nH.values), np.log10(df_gas.temp.values), df_elements, z, obs_data_dir, band = '0.5-2.0keV', fill_value = -400)) / df_gas.nH.values ** 2
+        df_gas['in_FOF'] = mask
+
+        df_elements = df_elements.div(df_elements.hydrogen, axis=0)
```

Comment:
@jemme07 it looks like you are only setting the mass fractions of sulphur and calcium after you used the elemental mass fractions in the X-ray interpolation on line 118. Is this correct? Right now the interpolation is using the same mass fraction as for silicon for these elements. Do the tables already assume the same sulphur/silicon and calcium/silicon?

Reply:
Yes, that is a standard assumption throughout EAGLE&co. Worth adding a comment though.

Reply:
Yes, as @MatthieuSchaller mentioned, this is a standard practice.